lib/geminabox/proxy/hostess.rb

```diff
@@ -49,22 +49,37 @@ def serve
       end
 
       get "/gems/*.gem" do
-        get_from_rubygems_if_not_local
+        get_from_external_sources_if_not_local
         serve
       end
 
       private
-      def get_from_rubygems_if_not_local
-
+      def get_from_external_sources_if_not_local
         file = File.expand_path(File.join(Server.data, *request.path_info))
-
         unless File.exists?(file)
-          ruby_gems_url = 'http://production.cf.rubygems.org'
-          path = File.join(ruby_gems_url, *request.path_info)
-          content = Geminabox.http_adapter.get_content(path)
+          content = false
+          Sources.external_sources.each do |source|
+            content = get_gem_from_external_source(source)
+            break if content
+          end
+
+          unless content
+            raise Geminabox::Error.new("Cant find ")
+          end
+
           GemStore.create(IncomingGem.new(StringIO.new(content)))
         end
+      end
 
+      def get_gem_from_external_source(source)
+        #external_source_url = 'http://production.cf.' + source
+        external_source_url = 'http://' + source
```

Comment:
This is what's tripping me up, isn't it? Can you just have the http:// or https:// set in the config file instead?